db/db.go

```diff
@@ -0,0 +1,110 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"strings"
+
+	_ "github.com/mattn/go-sqlite3" // sqlite
+	"go.uber.org/zap"
+)
+
+var logger = zap.NewExample().Sugar().Named("db")
+
+const defaultDBFile = "db.sqlite"
+
+type DB struct {
+	*sql.DB
+	file    string
+	cleanup func() error
+}
+
+// OpenDB opens sqlite database file.
+func OpenDB(file string) *DB {
+	if file == "" {
+		file = defaultDBFile
+	}
+	logger.Infow("opening sqlite database", "file", file)
+
+	stdDB, err := sql.Open("sqlite3", file)
+	if err != nil {
+		logger.Panic(err)
+	}
+
+	db := &DB{stdDB, file, func() error { return nil }}
+
+	_, err = db.Exec("PRAGMA journal_mode=WAL;")
+	if err != nil {
+		logger.Panic(err)
+	}
+
+	return db
+}
+
+// OpenTestDB generates a random database file name and opens it, returning cleanup function for use in tests.
+func OpenTestDB() *DB {
```

Comment:
you can use an in-memory sqlite db for tests. look at https://github.com/lbryio/morty/blob/master/store/store.go#L24

Reply:
Very good point, I had completely forgotten it can run in-memory.